sql/core/src/main/scala/org/apache/spark/sql/execution/r/MapPartitionsRWrapper.scala

```diff
@@ -25,6 +25,21 @@ import org.apache.spark.sql.Row
 import org.apache.spark.sql.types.{BinaryType, StructField, StructType}
 
 /**
+ * A function wrapper that applies the given R function to each partition of each group.
+ */
+private[sql] case class MapGroupsRWrapper(
```

Comment:
Instead, create a new physical operator, similar to MapGroupsExec. In the doExecute() of the new operator:
{code}
new RRunner
new Buffer for result
grouped.flatMap { case (key, rowIter) =>
   call RRunner.execute()
   append the values from the resulting iterator to the buffer
}
{code}